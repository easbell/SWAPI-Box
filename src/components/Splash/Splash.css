.splash {
  overflow: scroll;
  height: 200px;
  background-color: #545558;
  color: #fff;
  font-size: 22px;
  opacity: .7;
  width: 75%;
  padding: 25px;
  border-radius: 12px;
  margin: auto;
}