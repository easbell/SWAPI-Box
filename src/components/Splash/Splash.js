import React from 'react'
import './Splash.css'

const Splash = ({ randomFilm }) => {
  return (
    <div className='fade'>
      <div className='star-wars'>
        <div className='crawl'>
          <p>{randomFilm.openingCrawl}</p>
          <p>{randomFilm.title}</p>
          <p>{randomFilm.releaseDate}</p>
        </div>
      </div>
    </div>
  )
}

export default Splash;