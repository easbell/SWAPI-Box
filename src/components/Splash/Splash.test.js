import React, { Component } from 'react';
import Splash from './Splash';
import { shallow } from 'enzyme';

describe(Splash, () => {
  it('should match snapshot with all correct data passed through', () => {
    let mockRandomFilm = {
      openingCrawl: "It is a dark time for the Rebellion. Although the Death Star has been destroyed, Imperial troops have driven the Rebel forces from their hidden",
      releaseDate: "1980-05-17",
      title: "The Empire Strikes Back"}
      
    const wrapper = shallow(<Splash randomFilm={mockRandomFilm}/>)
    expect(wrapper).toMatchSnapshot();
  })
})