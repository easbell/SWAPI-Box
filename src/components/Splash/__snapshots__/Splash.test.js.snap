// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Splash should match snapshot with all correct data passed through 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Splash
    randomFilm={
      Object {
        "openingCrawl": "It is a dark time for the Rebellion. Although the Death Star has been destroyed, Imperial troops have driven the Rebel forces from their hidden",
        "releaseDate": "1980-05-17",
        "title": "The Empire Strikes Back",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          It is a dark time for the Rebellion. Although the Death Star has been destroyed, Imperial troops have driven the Rebel forces from their hidden
        </p>,
        <p>
          The Empire Strikes Back
        </p>,
        <p>
          1980-05-17
        </p>,
      ],
      "className": "splash",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "It is a dark time for the Rebellion. Although the Death Star has been destroyed, Imperial troops have driven the Rebel forces from their hidden",
        },
        "ref": null,
        "rendered": "It is a dark time for the Rebellion. Although the Death Star has been destroyed, Imperial troops have driven the Rebel forces from their hidden",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "The Empire Strikes Back",
        },
        "ref": null,
        "rendered": "The Empire Strikes Back",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "1980-05-17",
        },
        "ref": null,
        "rendered": "1980-05-17",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            It is a dark time for the Rebellion. Although the Death Star has been destroyed, Imperial troops have driven the Rebel forces from their hidden
          </p>,
          <p>
            The Empire Strikes Back
          </p>,
          <p>
            1980-05-17
          </p>,
        ],
        "className": "splash",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "It is a dark time for the Rebellion. Although the Death Star has been destroyed, Imperial troops have driven the Rebel forces from their hidden",
          },
          "ref": null,
          "rendered": "It is a dark time for the Rebellion. Although the Death Star has been destroyed, Imperial troops have driven the Rebel forces from their hidden",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "The Empire Strikes Back",
          },
          "ref": null,
          "rendered": "The Empire Strikes Back",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "1980-05-17",
          },
          "ref": null,
          "rendered": "1980-05-17",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
